dcrec/secp256k1/ciphering.go

```diff
@@ -23,18 +17,6 @@ var (
 	// corrupt ciphertext.
 	ErrInvalidMAC = errors.New("invalid mac hash")
```

Comment:
this should be defined as

```
var ErrInvalidMAC = v2.ErrInvalidMAC
```

currently both of those errors have different allocations and are unequal according to == and errors.Is.  Defining it this way instead would not break error detection.

Reply:
Nice catch.  Updated.